crates/router/src/db/business_profile.rs

```diff
@@ -109,13 +141,28 @@ impl BusinessProfileInterface for Store {
         merchant_id: &str,
     ) -> CustomResult<bool, errors::StorageError> {
         let conn = connection::pg_connection_write(self).await?;
-        storage::business_profile::BusinessProfile::delete_by_profile_id_merchant_id(
-            &conn,
-            profile_id,
-            merchant_id,
-        )
-        .await
-        .map_err(|error| report!(errors::StorageError::from(error)))
+        let db_func = || async {
+            storage::business_profile::BusinessProfile::delete_by_profile_id_merchant_id(
+                &conn,
+                profile_id,
+                merchant_id,
+            )
+            .await
+            .map_err(|error| report!(errors::StorageError::from(error)))
+        };
+
+        #[cfg(not(feature = "accounts_cache"))]
+        {
+            db_func().await
+        }
+
+        #[cfg(feature = "accounts_cache")]
+        {
+            let business_profile = self.find_business_profile_by_profile_id(profile_id).await?;
+            let result: bool = db_func().await?;
```

Comment:
We should delete it from cache first and then delete from db

Reply:
sure